### PR TITLE
[Android] ArrayList is faster than LinkedList for reasonable N.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/143721

Java LinkedList is just slooow.

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java [hunk 1]
```diff
@@ -735,7 +735,7 @@ private PerImageReader getActiveReader() {
           // Create a new ImageReader and add it to the queue.
           return getOrCreatePerImageReader(createImageReader());
         }
-        return imageReaderQueue.peekLast();
+        return imageReaderQueue.get(imageReaderQueue.size() - 1);
```
Comment (jonahwilliams) [inline shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java:738]: Need to check length here, the peek methods return null if the size isn't enough.

Comment (gmackall) [reply shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java:744]: Not that it much matters, but fun drive by comment to say you could use an `ArrayDeque` here instead if you wanted to keep the same convenience right? (they both implement the double ended queue interface)

Comment (jonahwilliams) [reply shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java:744]: That's a good idea, I will try that.

**Comment by chinmaygarde** [comment]
Are we making progress on this? Seems like a generally good thing to do.

**Comment by chinmaygarde** [comment]
Closing as stale.

**Pull request closed without merging**